include/magma/Deleter.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "magma/Device.hpp"
+
+namespace magma
+{
+  struct Deleter
+  {
+    Device<claws::no_delete> device;
+
+    template<class T>
+    void operator()(T const &obj) const
+    {
+      if (device)
+	device.destroy(obj);
```

Comment:
magma::Device::destroy is deleted. This causes compilation failure.
I see two possibilities:
- change it from deleted to private, and declare this class friend of it
- store a vk::Device instead, or cast to it when calling destroy()